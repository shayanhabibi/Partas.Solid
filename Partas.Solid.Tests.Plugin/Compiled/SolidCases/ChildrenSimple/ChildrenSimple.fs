module Partas.Solid.Tests.SolidCases.ChildrenSimple

open Partas.Solid
open Fable.Core

[<Erase>]
type TypeDefTest() =
    interface RegularNode

    [<SolidTypeComponent>]
    member props.typeDef = button () { "Button" }

[<Erase>]
type TypeDefTestWithProps() =
    interface RegularNode

    [<SolidTypeComponent>]
    member props.typeDef =
        button (class' = "ButtonClass", draggable = "indubitably") {
            "Button"
            span (class' = "sr-only") { "Span" }
        }

[<Erase>]
type TypeDefTestWithExtensions() =
    interface RegularNode

    [<SolidTypeComponent>]
    member props.typeDef =
        button(class' = "ButtonClass").attr ("Key", "Value") {
            "Button"
            span (class' = "sr-only") { "Span" }
            div(class' = "DivBelow").attr ("key", "value")
            "TextBelow"
        }
