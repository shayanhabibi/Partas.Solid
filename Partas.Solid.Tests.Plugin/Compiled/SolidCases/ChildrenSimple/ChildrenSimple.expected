import { splitProps } from "solid-js";

export function TypeDefTest(props) {
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, []);
    return <button>
        Button
    </button>;
}

export function TypeDefTestWithProps(props) {
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, []);
    return <button class="ButtonClass"
        draggable="indubitably">
        Button
        <span class="sr-only">
            Span
        </span>
    </button>;
}

export function TypeDefTestWithExtensions(props) {
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, []);
    return <button class="ButtonClass"
        Key="Value">
        Button
        <span class="sr-only">
            Span
        </span>
        <div class="DivBelow"
            key="value" />
        TextBelow
    </button>;
}

